### PR TITLE
Improve aggregation documentation

### PR DESCRIPTION
**Pull request by PointKernel**
## Description
This PR fixes several documentation issues uncovered while working on #16619. There are no actual code changes.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/include/cudf/detail/aggregation/aggregation.hpp [hunk 1]
```diff
@@ -1626,6 +1626,8 @@ struct dispatch_source {
  * parameter of the callable `F`
  * @param k The `aggregation::Kind` used to dispatch an `aggregation::Kind`
  * non-type template parameter for the second template parameter of the callable
+ * @param f The callable that accepts `data_type` and `aggregation::Kind` non-type template
+ * arguments.
```
Comment (harrism) [inline cpp/include/cudf/detail/aggregation/aggregation.hpp:1630]:
```suggestion
 * @param f The callable that accepts `data_type` and `aggregation::Kind` function object.
```

File: cpp/include/cudf/detail/aggregation/aggregation.hpp [hunk 2]
```diff
@@ -1644,8 +1646,8 @@ CUDF_HOST_DEVICE inline constexpr decltype(auto) dispatch_type_and_aggregation(d
  * @brief Returns the target `data_type` for the specified aggregation  k
  * performed on elements of type  source_type.
  *
- * aram source_type The element type to be aggregated
- * aram k The aggregation
+ * @param source_type The element type to be aggregated
+ * @param k The aggregation
```
Comment (harrism) [inline cpp/include/cudf/detail/aggregation/aggregation.hpp:1650]:
```suggestion
 * @param k The aggregation kind
```

**Review by harrism — COMMENTED** [review]
Some suggestions for clarification.

**Review by davidwendt — APPROVED** [review]

**Review by harrism — APPROVED** [review]

**Pull request merged**